src/MapViewController.swift

```diff
@@ -210,19 +210,20 @@ open class MapViewController: UIViewController, LocationManagerDelegate {
   private static let mapzenRights = "https://mapzen.com/rights/"
   private static let kGlobalPathApiKey = "global.sdk_mapzen_api_key"
   private static let kGlobalPathLanguage = "global.ux_language"
-  private let kDefaultSearchPinStyle = "{ style: sdk-point-overlay, sprite: ux-search-active, size: [24, 36px], collide: false, interactive: true }"
+
   private var isCurrentlyVisible = false // We can't rely on things like window being non-nil because page controllers have a non-nil window as they're being rendered off screen
 
   let application : ApplicationProtocol
   open var tgViewController: TGMapViewController = TGMapViewController()
-  var currentLocationGem: TGMarker?
+  var currentLocationGem: Marker?
   var lastSetPoint: TGGeoPoint?
   var shouldShowCurrentLocation = false
-  var currentRouteMarker: TGMarker?
+  var currentRouteMarker: Marker?
   var currentRoute: OTRRoutingResult?
   open var shouldFollowCurrentLocation = false
   open var findMeButton = UIButton(type: .custom)
   var currentAnnotations: [PeliasMapkitAnnotation : TGMarker] = Dictionary()
+  var currentMarkers: [TGMarker : Marker] = Dictionary()
```

Comment:
Agreed that this could easily go away, I think a good pattern here would be to be able to set a `userData` directly to a `TGMarker`:

```objc
- (void)userData:(void*)userData
```

so that the user could set whatever data it needs to it, and retrieve it in a callback whenever a selection happens, so storing objects mapping like this wouldn't be necessary.